src/intersections/raytriangle.jl

```diff
@@ -49,7 +56,15 @@ function intersection(f, r::Ray{3,T}, t::Triangle{3,T}) where {T}
     return @IT NoIntersection nothing f
   end
 
+  any(o .≈ vs) && (return @IT CornerOriginTouchingRayTriangle r(λ) f)
+
+  λ ≈ zero(T) && (return @IT EdgeOriginTouchingRayTriangle r(λ) f)
+
+  coords = (u, v, det - u - v)
+  count(x -> x ≈ zero(T), coords) == 1 && (return @IT EdgeTouchingRayTriangle r(λ) f)
+  count(x -> x ≈ det, coords) == 1 && (return @IT CornerTouchingRayTriangle r(λ) f)
```

Comment:
I think we can make coords a Vector and then use isapprox as it is done in other places in the codebase.